instrumentation/message-handler/library/src/main/java/io/opentelemetry/instrumentation/messagehandler/BatchMessageHandler.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.messagehandler;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.TracerBuilder;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.contrib.awsxray.propagator.AwsXrayPropagator;
+import io.opentelemetry.instrumentation.api.instrumenter.messaging.MessageOperation;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public abstract class BatchMessageHandler<T> {
+  private static final Logger logger = Logger.getLogger(BatchMessageHandler.class.getName());
+  private static final String AWS_TRACE_HEADER_PROPAGATOR_KEY = "x-amzn-trace-id";
+
+  protected String messagingOperation;
+  protected OpenTelemetry openTelemetry;
+  protected String spanName;
+
+  public BatchMessageHandler(OpenTelemetry openTelemetry) {
+    this(openTelemetry, MessageOperation.RECEIVE.name());
+  }
+
+  public BatchMessageHandler(OpenTelemetry openTelemetry, String messageOperation) {
+    this(openTelemetry, messageOperation, "Batch Message");
+  }
+
+  public BatchMessageHandler(
+      OpenTelemetry openTelemetry, String messageOperation, String spanName) {
+    this.openTelemetry = openTelemetry;
+    this.spanName = spanName;
+    this.messagingOperation = messageOperation;
+  }
+
+  public abstract String getParentHeaderFromMessage(T t);
+
+  protected abstract void doHandleMessages(Collection<T> messages);
+
+  protected void addMessagingAttributes(SpanBuilder spanBuilder) {
+    spanBuilder.setAttribute(SemanticAttributes.MESSAGING_OPERATION, messagingOperation);
+  }
+
+  public void handleMessages(Collection<T> messages) {
+    TracerBuilder tracerBuilder =
+        openTelemetry
+            .tracerBuilder("io.opentelemetry.message.handler")
+            .setInstrumentationVersion("1.0");
+
+    Span parentSpan = Span.current();
+
+    SpanBuilder spanBuilder =
+        tracerBuilder
+            .build()
+            .spanBuilder(spanName)
+            .setParent(Context.current().with(parentSpan))
```

Comment:
I think this is a no-op, as the current span will be the implicit default parent anyways.